app/versioning.py

```diff
@@ -0,0 +1,62 @@
+from typing import Callable, List
+
+import flask
+from werkzeug.exceptions import BadRequest
+
+VALID_API_VERSIONS = ['1.0']
+"""API versions that are valid for a user to request.
+
+NOTE: the first value in this list should be the latest supported version, but the
+ordering of subsequent version numbers is not important.
+"""
+LATEST_API_VERSION = VALID_API_VERSIONS[0]
+"""latest API version to default to when none other is requested"""
+
+
+def versioned(valid_versions: List[str] = None, throw_on_invalid: bool = True):
+    """
+    Extracts requested API version from X-API-Version HTTP header and passes
+    to wrapped function in the `version` kwarg.
+
+    :param throw_on_invalid: whether to throw exception on invalid passed version,
+        defaults to `True`
+    :param valid_versions: list of API versions to consider as valid, falls back to
+        default list of valid API versions when not specified
+    :raise InvalidApiVersion: when `throw_on_invalid` is `True` and invalid API version
+        was specified in the client request
+    """
+
+    def decorator(function: Callable):
+        def wrapper(*args, **kwargs):
+            if flask.has_request_context():
+                version = flask.request.headers.get('x-api-version', LATEST_API_VERSION)
+                if version not in valid_versions:
+                    if throw_on_invalid:
+                        raise InvalidApiVersion(version)
+                    else:
+                        version = LATEST_API_VERSION
+            else:
+                version = LATEST_API_VERSION
+
+            kwargs['version'] = float(version)
```

Comment:
We talked about keeping the versions a list to support other arbitrary strings such as `1.2-slack`. If you pass that into float you get something like this:
```python
>>> x = '1.2-slack'
>>> float(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: could not convert string to float: '1.2-slack'
```

Is there a problem with just keeping it a string? Maybe we could do `str(version)` instead?

Reply:
Yeah, I’m groovy with making it back into a string.

Reply:
- [ ] change version from being converted to a float to being maintained as a string